src/thread/__wait.c

```diff
@@ -2,6 +2,7 @@
 
 void __wait(volatile int *addr, volatile int *waiters, int val, int priv)
 {
+#ifndef __CHEERP__
```

Comment:
What's the idea here? In a multithreaded future it will be actually supported

Reply:
These are mostly hacks after looking at the compiled code, since wait will possibly not be eliminated form the code, and then having a call to the dummy __syscall_futex means it will throw at execution time.

There are other ways out (eg. providing basic implementation of __syscall_futex) that are probably safer. In any case both this and locks commit can be safely dropped / reconsidered in the future.